### PR TITLE
Drop old given syntax

### PR DESCRIPTION
**Pull request by odersky**
Based on #10598

### CODE REVIEW

File: library/src/scala/quoted/Quotes.scala [hunk 1]
```diff
@@ -246,7 +246,7 @@ trait Quotes { self: runtime.QuoteUnpickler & runtime.QuoteMatching =>
     type PackageClause <: Tree
 
     /** `TypeTest` that allows testing at runtime in a pattern match if a `Tree` is a `PackageClause` */
-    given PackageClauseTypeTest as TypeTest[Tree, PackageClause] = PackageClauseTypeTestImpl
+    given PackageClauseTypeTest: TypeTest[Tree, PackageClause] = PackageClauseTypeTestImpl
 
     /** Implementation of `PackageClauseTypeTest` */
     protected val PackageClauseTypeTestImpl: TypeTest[Tree, PackageClause]
```
Comment (smarter) [inline library/src/scala/quoted/Quotes.scala:252]: Looks like you could directly use abstract givens for this now:
```suggestion
    given PackageClauseTypeTest: TypeTest[Tree, PackageClause]
```

Comment (smarter) [reply library/src/scala/quoted/Quotes.scala:252]: @nicolasstucki ^

Comment (nicolasstucki) [reply library/src/scala/quoted/Quotes.scala:252]: Is that supported now?

Comment (nicolasstucki) [reply library/src/scala/quoted/Quotes.scala:252]: If it is supported I will create a PR to simplify this.

**Review by nicolasstucki — APPROVED** [review]

**Comment by nicolasstucki** [comment]
@odersky I merged the current version to avoid conflicts with other PRs.

**Pull request merged**